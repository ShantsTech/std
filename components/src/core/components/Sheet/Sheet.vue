<!--
/*
 * This file is part of Shants Tech LLC
 *
 * Copyright (C) 2020 Shants Tech LLC
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */
-->

<template>
  <div :class="classes">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import {SheetType, TYPES, TYPE_WHITE} from './types';

export default defineComponent({
  name: 'std-sheet',

  props: {
    type: {
      type: String,
      default: TYPE_WHITE,
      validator: function(value: SheetType) {
        return TYPES.indexOf(value) !== -1;
      },
    },
    rounded: {
      type: Boolean,
      default: true,
    },
    gutters: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    classes(): object {
      return {
        'std-sheet': true,
        'std-sheet--rounded': this.rounded,
        'std-sheet--gutters': this.gutters,
        [`std-sheet--${this.type}`]: true,
      };
    },
  },
});
</script>

<style src="./sheet.scss" lang="scss" scoped></style>
