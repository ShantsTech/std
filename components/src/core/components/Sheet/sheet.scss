// This file is part of Shants Tech LLC
//
// Copyright (C) 2020 Shants Tech LLC
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import '../../../styles';
@import 'variables';

.std-sheet {
  background-color: $std-sheet-white-color;

  &--gray-darken-2 {
    background-color: $std-sheet-gray-darken-2-color;
  }
  &--gray-darken-1 {
    background-color: $std-sheet-gray-darken-1-color;
  }
  &--gray {
    background-color: $std-sheet-gray-color;
  }
  &--gray-lighten-1 {
    background-color: $std-sheet-gray-lighten-1-color;
  }
  &--gray-lighten-2 {
    background-color: $std-sheet-gray-lighten-2-color;
  }
  &--light-gray {
    background-color: $std-sheet-light-gray-color;
  }
  &--pastel-white {
    background-color: $std-sheet-pastel-white-color;
  }
  &--white-shadow {
    background-color: $std-sheet-white-shadow-color;
  }

  &--rounded {
    border-radius: $std-sheet-border-radius;
  }

  &--gutters {
    padding-top: $std-sheet-padding-vertical;
    padding-bottom: $std-sheet-padding-vertical;
    padding-left: $std-sheet-padding-horizontal;
    padding-right: $std-sheet-padding-horizontal;
  }
}
