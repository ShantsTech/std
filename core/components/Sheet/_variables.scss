// This file is part of Stants Tech LLC
//
// Copyright (C) 2020 Stants Tech LLC
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

/* OXD - Interface Colors */
$std-sheet-gray-darken-2-color: $std-interface-gray-darken-2-color !default;
$std-sheet-gray-darken-1-color: $std-interface-gray-darken-1-color !default;
$std-sheet-gray-color: $std-interface-gray-color !default;
$std-sheet-gray-lighten-1-color: $std-interface-gray-lighten-1-color !default;
$std-sheet-gray-lighten-2-color: $std-interface-gray-lighten-2-color !default;

/* OXD - Background & Overly Colors */
$std-sheet-light-gray-color: $std-background-light-gray-color !default;
$std-sheet-pastel-white-color: $std-background-pastel-white-color !default;
$std-sheet-white-shadow-color: $std-background-white-shadow-color !default;

$std-sheet-white-color: $std-white-color !default;

$std-sheet-border-radius: $std-border-radius !default;
$std-sheet-padding-vertical: 25px !default;
$std-sheet-padding-horizontal: 25px !default;
