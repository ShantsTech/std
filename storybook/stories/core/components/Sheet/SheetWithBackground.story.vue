<!--
/*
 * This file is part of Shants Tech LLC
 *
 * Copyright (C) 2020 Shants Tech LLC
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */
-->
-->

<template>
  <std-sheet type="pastel-white" :rounded="false">
    <std-sheet v-bind="$attrs">
      <std-text type="card-title">Card Title</std-text>
      <div :style="{width: '100px', height: '100px'}"></div>
    </std-sheet>
  </std-sheet>
</template>

<script>
import Sheet from '@sthrm/std/core/components/Sheet/Sheet';
import Text from '@sthrm/std/core/components/Text/Text';

export default {
  components: {
    'std-sheet': Sheet,
    'std-text': Text,
  },
  inheritAttrs: false,
};
</script>
