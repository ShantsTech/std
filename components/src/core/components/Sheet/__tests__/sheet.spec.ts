/*
 * This file is part of Shants Tech LLC
 *
 * Copyright (C) 2020 Shants Tech LLC
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */

import {mount} from '@vue/test-utils';
import Sheet from '@ohrm/std/core/components/Sheet/Sheet.vue';

describe('Sheet.vue', () => {
  it('should renders STD Sheet', () => {
    const wrapper = mount(Sheet, {
      props: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, not rounded', () => {
    const wrapper = mount(Sheet, {
      props: {rounded: false},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, without gutters', () => {
    const wrapper = mount(Sheet, {
      props: {gutters: false},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type white', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'white'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type white-shadow', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'white-shadow'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type pastel-white', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'pastel-white'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type light-gray', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'light-gray'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type gray-lighten-2', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-lighten-2'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type gray-lighten-1', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-lighten-1'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type gray', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type gray-darken-1', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-darken-1'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD Sheet, type gray-darken-2', () => {
    const wrapper = mount(Sheet, {
      props: {type: 'gray-darken-2'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
