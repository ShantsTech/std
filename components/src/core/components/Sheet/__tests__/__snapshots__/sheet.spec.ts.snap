// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sheet.vue should renders STD Sheet 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--white"></div>`;

exports[`Sheet.vue should renders STD Sheet, not rounded 1`] = `<div class="std-sheet std-sheet--gutters std-sheet--white"></div>`;

exports[`Sheet.vue should renders STD Sheet, type gray 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--gray"></div>`;

exports[`Sheet.vue should renders STD Sheet, type gray-darken-1 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--gray-darken-1"></div>`;

exports[`Sheet.vue should renders STD Sheet, type gray-darken-2 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--gray-darken-2"></div>`;

exports[`Sheet.vue should renders STD Sheet, type gray-lighten-1 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--gray-lighten-1"></div>`;

exports[`Sheet.vue should renders STD Sheet, type gray-lighten-2 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--gray-lighten-2"></div>`;

exports[`Sheet.vue should renders STD Sheet, type light-gray 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--light-gray"></div>`;

exports[`Sheet.vue should renders STD Sheet, type pastel-white 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--pastel-white"></div>`;

exports[`Sheet.vue should renders STD Sheet, type white 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--white"></div>`;

exports[`Sheet.vue should renders STD Sheet, type white-shadow 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--gutters std-sheet--white-shadow"></div>`;

exports[`Sheet.vue should renders STD Sheet, without gutters 1`] = `<div class="std-sheet std-sheet--rounded std-sheet--white"></div>`;
